### PR TITLE
feat: permit2 animations WEB-2036

### PR DESCRIPTION
**Pull request by just-toby**
## Description

implement the animations between swap submission flow steps

_Relevant docs:_ https://www.figma.com/file/ksWAw6EZiiMuRyMQQHicoH/Permit2-Revisions?type=design&node-id=531-36256&t=Xm9bH0VJ03E813sU-0

## Screen capture

https://github.com/Uniswap/interface/assets/66155195/3c304617-837e-46e8-8725-a44753be3423

https://github.com/Uniswap/interface/assets/66155195/b149ad80-b768-4f22-bc65-9e77dc5c22fd

- [x] N/A

### Automated testing

- [x] Unit test
- [x] Integration/E2E test

### CODE REVIEW

**Comment by grabbou** [comment]
Putting review on hold once we merged the other modal-related work and rebased for clarity.

File: src/components/swap/PendingModalContent/ErrorModalContent.tsx
Comment (just-toby) [inline src/components/swap/PendingModalContent/ErrorModalContent.tsx:12]: this whole file is just copied from the previous `src/components/swap/PendingModalContent.tsx`

File: src/components/swap/PendingModalContent/Logos.tsx
Comment (just-toby) [inline src/components/swap/PendingModalContent/Logos.tsx:7]: copied from `src/components/swap/PendingModalContent.tsx`

Comment (tinaszheng) [reply src/components/swap/PendingModalContent/Logos.tsx:6]: can this be an absolute import instead of relative 😅 the number of `../` is giving me an aneurysm

File: src/components/swap/PendingModalContent/Logos.tsx
Comment (tinaszheng) [inline src/components/swap/PendingModalContent/Logos.tsx:40]:
```suggestion
  currency?: Currency
```

File: src/components/swap/PendingModalContent/Logos.tsx
Comment (tinaszheng) [inline src/components/swap/PendingModalContent/Logos.tsx:65]: so there are a bunch of components in here that have the same property of opacity 1 <-> 0 with a transition timing. Thoughts on making a general `FadeIn` component that is a simple div wrapper with these two properties?

Comment (tinaszheng) [reply src/components/swap/PendingModalContent/Logos.tsx:65]: im wondering also if this component should handle mounting/unmounting children when they're done with the exit animation and no longer visible. you could probably make use of `useUnmountingAnimation` to handle that logic - that way we wouldn't have to actually mount all these elements on the page if they're not actually rendered. you could then amend your top-level logic to be like:

```
{currentStep === ConfirmModalState.APPROVING_TOKEN && <PaperIcon />}
```

File: cypress/e2e/permit2.test.ts
Comment (zzmp) [inline cypress/e2e/permit2.test.ts:79]: `cy.wait('@eth_signTyped_date_v4')` should work here, and not require a spy.

File: src/components/swap/PendingModalContent/Logos.tsx
Comment (JackShort) [inline src/components/swap/PendingModalContent/Logos.tsx:119]: does this need to be a fragment?

File: src/components/swap/PendingModalContent/Logos.tsx
Comment (JackShort) [inline src/components/swap/PendingModalContent/Logos.tsx:53]: should this not be this: 
```
display ${({ theme }) => `${theme.transition.duration.medium} ${theme.transition.timing.inOut}`}
```

**Review by zzmp — APPROVED** [review]

File: src/components/swap/PendingModalContent/index.tsx
Comment (JackShort) [inline src/components/swap/PendingModalContent/index.tsx:73]: same question about this here

```
  display ${({ theme }) => `${theme.transition.duration.medium} ${theme.transition.timing.inOut}`}
```

**Review by JackShort — APPROVED** [review]
lgtm - few small questions

**Review by cartcrom — APPROVED** [review]

**Review by zzmp — APPROVED** [review]

**Pull request merged**
